lib/explorer/data_frame.ex

```diff
@@ -1327,15 +1327,20 @@ defmodule Explorer.DataFrame do
     * `:config` - An optional struct, keyword list or map, normally associated with remote
       file systems. See [IO section](#module-io-operations) for more details. (default: `nil`)
 
+    * `:streaming` - Tells the backend if it should use streaming, which means
+      that the dataframe is not loaded to the memory at once, and instead it is
+      written in chunks from a lazy dataframe.  This option is not supported when using an S3
+      entry.
+
   """
   @doc type: :io
   @spec to_csv(df :: DataFrame.t(), filename :: fs_entry() | String.t(), opts :: Keyword.t()) ::
           :ok | {:error, Exception.t()}
   def to_csv(df, filename, opts \\ []) do
-    opts = Keyword.validate!(opts, header: true, delimiter: ",", config: nil)
+    opts = Keyword.validate!(opts, header: true, delimiter: ",", streaming: false, config: nil)
```

Comment:
We should probably default to true as in the other operations.

Reply:
@josevalim - Will do!  Do you think it's worth mentioning in the docs that streaming set to true for a DataFrame with an S3 entry will fail explicitly?  I could also design it so that the streaming option is ignored for a LazyFrame with an S3 entry, but for now I used the same behavior as `to_parquet/3` prior to having streaming implemented.

Reply:
@ryancurtin the most important is to default to true, so I would say "defaults to true on supported filesystems, ignored on all others".

Reply:
Sounds good, I just added that and tweaked the behavior so as not to raise an error when using the streaming option on an S3 filesystem given that `true` is now the default.  Tests are also updated to reflect that and try both cases.